File: react/ml-runner/src/services/RunnerService.ts
import axios from "axios";
import authorizationHeader from "../services/AuthorizationHeader";
import {BACKEND_URL} from "../helpers/url";
import {HyperParameter} from "../helpers/types";
const API_URL = BACKEND_URL + "/api";

/**
 * Providing the access to the API for exeution of a runner, monitoring its status and getting results.
 */
const run = (projectId: number, hyperParameters: HyperParameter[]) =>
{
    return axios.post<any>(API_URL + '/project/runner/run', { projectId: projectId, hyperParameters: hyperParameters }, { headers: authorizationHeader() });
}

const getStatus = (runnerId: number) =>
{
    return axios.get<any>(API_URL + '/project/runner/status?runnerId=' + runnerId, { headers: authorizationHeader() });
}

const getResult = (projectId: number, runnerId: number) => {
    return axios.get<any>(BACKEND_URL + '/api/project/runner/result?projectId=' + projectId + '&' + 'runnerId=' + runnerId, {headers: authorizationHeader()})
}

export default { run, getStatus, getResult }

File: java/web/src/main/java/com/jan/web/runner/RunnerService.java
package com.jan.web.runner;

import com.jan.web.docker.ContainerEntity;
import com.jan.web.project.Project;
import com.jan.web.runner.parameter.HyperParameter;
import com.jan.web.runner.result.Result;
import com.jan.web.runner.status.RunnerStatus;
import com.jan.web.security.user.User;
import org.json.JSONException;

import java.io.IOException;
import java.util.List;
import java.util.Optional;

/**
 * Interface proving methods for execution of a {@link Runner} of {@link Project},
 * monitoring execution status and providing results.
 */
public interface RunnerService
{
   void runProject(List<HyperParameter> hyperParameters, Project project, ContainerEntity containerEntity);
   RunnerStatus getStatus(long containerId, long runnerId) throws JSONException, IOException;
   Optional<Result> getResult(long containerId, long projectId, long runnerId) throws IOException, JSONException;
   boolean isAnyRunnerRunning(User user);
}
